lib/publiccloud/instance.pm

```diff
@@ -46,10 +48,17 @@ sub run_ssh_command {
 
     $args{timeout} //= SSH_TIMEOUT;
 
+    my $cmd = $args{cmd};
+    delete($args{cmd});
+    unless ($args{no_quote}) {
+        $cmd =~ s/'/'"'"'/g;
+        $cmd = "'$cmd'";
+    }
+    delete($args{no_quote});
+
     my $ssh_cmd = sprintf('ssh %s -i "%s" "%s@%s" -- %s',
-        $self->ssh_opts, $self->ssh_key, $self->username, $self->public_ip, $args{cmd});
+        $self->ssh_opts, $self->ssh_key, $self->username, $self->public_ip, $cmd);
```

Comment:
Actually I tried to keep the dependencies as small as possible with file-system configurations. 
But I have no problem with this. I think if someone wanna use `.ssh/config` instead of he can just overwrite this ssh_opts with
```
$instance->ssh_opts("");
```

But you also mentioned in PR#8422 that you need other options. What was you talking about explicit? 
What I currently see is, that we are not able to set a different username <= I will add this.

Reply:
In #8422 in `ssh_init.pm` on line 52 I use:
```
script_run("ssh -t -R $upload_port:$upload_host:$upload_port " . $instance->public_ip . " 'rm -rf /dev/sshserial; mkfifo -m a=rwx /dev/sshserial; tail -Fn +1 /dev/sshserial' | tee /dev/$serialdev ", 0);
```

Mind also the zero timeout at the end - this means the script will not be waiting for the command and the command will actually keep running until the last test module `ssh_end.pm` where I kill it.

Reply:
I see. It should work with:
```
$instance->ssh_opts("-t -R $upload_port:$upload_host:$upload_port");
$instance->run_ssh_command('sudo su; rm -rf /dev/sshserial; mkfifo -m a=rwx /dev/sshserial; tail -Fn +1 /dev/sshserial' | tee /dev/$serialdev ", timeout => 0, no_quote => 1);
```
But it's not more readable then yours. So lets keep this special thing as it is, or?

Reply:
> What I currently see is, that we are not able to set a different username <= I will add this.

Please do that :smile: 

> But it's not more readable then yours. So lets keep this special thing as it is, or?

I actually find it more readable

Reply:
ok I did. With latest version it should be:

```
$instance->run_ssh_command(
    cmd => 'rm -rf /dev/sshserial; mkfifo -m a=rwx /dev/sshserial; tail -Fn +1 /dev/sshserial' | tee /dev/$serialdev ", 
    timeout => 0, 
    no_quote => 1, 
    ssh_opts => '-t -R $upload_port:$upload_host:$upload_port', 
    username => 'root'
);